guide/docs/popular-topics/permissions.mdx

```diff
@@ -1,8 +1,382 @@
+-   First we apply the `@everyone` role's permissions.
+-   Then, for each role the member has we apply it's permissions too.
+-   -   If at this point the member has the "Administrator" permission, all others are granted automatically
+        (the member is a guild-wide administrator, and, as such, is functionally equivalent to the owner except
+        they can't delete the guild).
+-   After that, if we're in a channel, the following is additionally done.
+-   -   We apply it's overwrites in the same order: `@everyone` overwrites,
+        role overwrites and finally member overwrites.
+-   -   Then [channel-specific implicit permissions](#channel-specific-implicit-permissions) are handled.
+-   In the end, we take into account [guild-wide implicit permissions](#guild-wide-implicit-permissions).
+
+In case of calculating permissions for a role, the strategy is the same as if we were calculating permissions
+for a member with only that role (and `@everyone`), except that member overwrites are
+not taken into account.
+
+:::note
+Curious about the implementation? You can see it [here](https://git.disnake.dev/disnake/blob/master/disnake/abc.py#L651-L808).
+:::
+
+### Implicit permissions
+
+In Discord, there are some permissions that, when revoked, basically make it impossible to utilize other
+granted permissions. For example, take a user with two permissions: "View Channel" and
+"Send Messages". If at some point user loses the "View Channel" permission, it wouldn't matter whether
+they can send messages in that channel: they just can't see the channel in the UI. disnake's permission system
+attempts to handle such cases via "implicit permissions" - permissions that, when granted or, more commonly,
+revoked, automatically grant (revoke) other permissions as well. The currently handled cases are separated
+into two parts: guild-wide rules and channel-specific rules, and are described below.
+
+#### Guild-wide implicit permissions
+
+Guild-wide implicit permissions currently consist of a single rule.
+
+1. Timed out members are spectators.
+
+This rule autmatically revokes all permissions for timed out members _except_ "View Channel" and
+"Read Message History", which are left [untouched](#untouched-permissions).
+
+#### Channel-specific implicit permissions
+
+Channel-specific implicit permissions currently include two rules.
+
+1. If you can't send a message in a channel, then you can't sent _any_ message in it.
+
+This rule automatically revokes "Send TTS Messages", "Send Voice Messages", "Mention Everyone", "Embed Links"
+and "Attach Files" permissions for members that do not have the "Send Messages" permission.
+
+2. If you can't view a channel, they you can't do anything in it.
+
+This rule automatically revokes "All Channel" permissions, as specified by <DocsLink reference="disnake.Permissions.all_channel" />,
+if the member doesn't have the "View Channel" permission.
```

Comment:
I'm not sure if we want to document all of these - Discord doesn't either (it is, and I quote, ["based on logical conclusions"](https://discord.com/developers/docs/topics/permissions#implicit-permissions)).
If anything, mentioning that this is not an exhaustive list would make sense, as there are a bunch of fun edge cases

Reply:
I rephrased this a bit so that these rules only act as examples, and by no means are an exhaustive list of all handled/possible cases where implicit permissions come into action. 537115d0ef62104ffc522159944da587cc554de6